src/data/equiv/algebra.lean

```diff
@@ -355,4 +380,112 @@ instance symm.is_ring_hom {e : α ≃r β} : is_ring_hom e.to_equiv.symm := hom
 @[simp] lemma to_equiv_symm_apply (e : α ≃r β) (x : β) :
   e.symm.to_equiv x = e.to_equiv.symm x := rfl
 
+/-- Two ring isomorphisms agree if they are defined by the
+    same underlying function. -/
+@[extensionality] lemma ext {R S : Type*} [ring R] [ring S]
+  {f g : ring_equiv R S} (h : f.to_fun = g.to_fun) : f = g :=
+begin
+  have h₁ := @equiv.eq_of_to_fun_eq _ _ f.to_equiv g.to_equiv h,
+  cases f, cases g, congr,
+  { exact h },
+  { exact congr_arg equiv.inv_fun h₁ }
+end
+
 end ring_equiv
+
+/-
+We define the type of automorphisms on groups, additive monoids,
+additive groups and rings using `mul_equiv`, `add_equiv` and `ring_equiv`.
+In each case this type also forms a group.
+-/
+
+namespace monoid
+
+def aut (γ : Type u) [has_mul γ] := mul_equiv γ γ
```

Comment:
Why do you only ask for `has_mul` here, wheres you ask for `group` in the group-version below.

Reply:
Hmm... So what should we do here? I changed this to just requiring `has_mul` in the group version, but of course now `monoid.aut` and `group.aut` are identical. Where should this notion live?

Reply:
Changed to `mul_aut`; requires only `[has_mul α]`.